### PR TITLE
[master < T0728-MG]  Add docs for temporal types

### PR DESCRIPTION
**Pull request by kostasrim**

### CODE REVIEW

**Review by gitbuda — CHANGES_REQUESTED** [review]
@kostasrim please update the changelog!

File: docs/reference-guide/temporal_types/overview.md [hunk 1]
```diff
@@ -0,0 +1,238 @@
+---
+id: overview
+title: Temporal types
+sidebar_label: Temporal types overview
+slug: /reference-guide/temporal_types/
+---
+
+Temporal types can be stored as properties of a graph. The user can use cypher queries or write/read procedures to write/read these properties. The following table summarizes the temporal types that memgraph currently supports:
```
Comment (antaljanosbenjamin) [inline docs/reference-guide/temporal_types/overview.md:8]:
```suggestion
Temporal types can be stored as properties of a vertex or edge. The user can use cypher queries or write/read procedures to write/read these properties. The following table summarizes the temporal types that memgraph currently supports:
```

File: docs/reference-guide/temporal_types/overview.md [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+Note, that all the types excluding Durations are ISO 8601 compliant
+
+## Duration
+You can create `Duration` from a string or a map by calling the function `Duration`.
+
+For strings, the duration format is:
+
+`P[nD]T[nH][nM][nS]` n stands for number which can be a fraction  and the capital letters are used as separator with each field in `[]` being optional. For strings, Memgraph only allows one field to be of type double. However, for maps, every field can be a double, an int or a mixture of both. Note, we support negative durations.
+
+Symbol table
```
Comment (antaljanosbenjamin) [inline docs/reference-guide/temporal_types/overview.md:26]: I would either convert this to a sub-heading or remove it. The same true for `Duration(/date/locatime/localdatetime) properties`

File: docs/reference-guide/temporal_types/overview.md [hunk 3]
```diff
@@ -0,0 +1,238 @@
...
+:-:|:-:
+day|This converts all the microseconds to days and returns them.|/
+hour|This subtracts days and returns the leftover as hours.|/
+minute|This subtracts the days and returns the leftover as minutes.|/
+second|This subtracts the days and returns the leftover as seconds.|/
+millisecond|This subtracts the days and returns the leftover as milliseconds.|/
+microsecond|This subtracts the days and returns the leftover as microseconds.|/
+nanosecond|This subtracts the days and returns the leftover as nanoseconds.|/
+
+Example
```
Comment (antaljanosbenjamin) [inline docs/reference-guide/temporal_types/overview.md:66]:
```suggestion
Example:
```

File: docs/reference-guide/temporal_types/overview.md [hunk 4]
```diff
@@ -0,0 +1,238 @@
...
+
+`YYYY-MM-DD` or `YYYYMMDD` or `YYYY-MM`
+
+Symbol table:
+
+name|description
+:-:|:-:
+Y|Year|/
+M|Month|/
+S|Second|/
```
Comment (antaljanosbenjamin) [inline docs/reference-guide/temporal_types/overview.md:82]:
```suggestion
D|Date|/
```

File: docs/reference-guide/temporal_types/overview.md [hunk 5]
```diff
@@ -0,0 +1,238 @@
...
+
+The lowest year is `0` and the highest is `9999`.
+
+Example:
+```cypher
+CREATE (:Birthdays {Arnold : date("1947-07-30")})
```
Comment (antaljanosbenjamin) [inline docs/reference-guide/temporal_types/overview.md:88]:
```suggestion
CREATE (:Person {birthday : date("1947-07-30")})
```
This make a bit more sense in my opinion, but feel free to leave as is.

File: docs/reference-guide/temporal_types/overview.md [hunk 6]
```diff
@@ -0,0 +1,238 @@
...
+
+Temporal types support native arithmetic and the operations are summarized in the following table:
+
+Duration operations: 
+
+op|result
+:-:|:-:
+duration + duration|duration|/
+duration - duration|duration|/
+-duration|negation|/
```
Comment (antaljanosbenjamin) [inline docs/reference-guide/temporal_types/overview.md:205]:
```suggestion
-duration|duration|/
```

Eveywhere else the result is a type, therefore the negation feels strange to me.

File: docs/reference-guide/temporal_types/overview.md [hunk 7]
```diff
@@ -0,0 +1,238 @@
...
+
+Date operations: 
+
+op|result
+:-:|:-:
+date + duration|date|/
+duration + date|date|/
```
Comment (antaljanosbenjamin) [inline docs/reference-guide/temporal_types/overview.md:212]: Why the `duration + x` is only present for dates? If I am not mistaken this should either work for everything else or non of them.

File: docs/reference-guide/temporal_types/overview.md [hunk 8]
```diff
@@ -0,0 +1,238 @@
...
+localdatetime + duration|localdatetime|/
+localdatetime - duration|localdatetime|/
+localdatetime - localdatetime|duration|/
+
+## Procedures API
+Temporal types can be used with query procedures. You can find the documentation in the following pages:
+
+[C API](/reference-guide/query-modules/api/c-api.md)
+
+[Pyhon API](/reference-guide/query-modules/api/python-api.md)
```
Comment (antaljanosbenjamin) [inline docs/reference-guide/temporal_types/overview.md:237]:
```suggestion
Temporal types can also be used within query modules, you can find the regarding documentation on the [Pyhon API](/reference-guide/query-modules/api/python-api.md) and [C API](/reference-guide/query-modules/api/c-api.md) pages.
```

**Review by gitbuda — APPROVED** [review]
Not to block here :D

**Review by antaljanosbenjamin — APPROVED** [review]

**Review by g-despot — APPROVED** [review]

**Pull request merged**